tycho-extras/tycho-pomless/src/main/java/org/eclipse/tycho/pomless/AbstractTychoMapping.java

```diff
@@ -162,6 +161,10 @@ protected File getRealArtifactFile(File polyglotArtifactFile) {
         return polyglotArtifactFile;
     }
 
+    File getArtifactDirectory(Map<String, ?> options) {
```

Comment:
I think we should not use this method here in the abstract class as on some places as we specifically want to support "directory poms". instead I think it would be better to move Reading-code of the build properties into `getEnhancementProperties` and instead have a (non static) `getBuildProperties` method that returns the location of the file.